flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,204 @@
+                            cert_required = ssl.CERT_REQUIRED
+                        else:
+                            cert_required = ssl.CERT_NONE
+                        logger.verbose('TLS cert_required={}'.format(str(cert_required)))
+
+                        self.tls_set(ca_certs=certs.get('broker_ca_cert',None),
+                                      certfile=certs.get('client_cert',None),
+                                      keyfile=certs.get('client_key',None),
+                                      cert_reqs=cert_required,
+                                      tls_version=tls_version)
+
+                        if not certs.get('validate_broker_cert',None):
+                            self.tls_insecure_set(True)
+
+                            logger.debug('TLS insecure cert mode enabled. Broker cert will not be validated')
+                        else:
+                            logger.debug('TLS secure cert mode enabled. Broker cert will be validated')
+
+                    #Handle user/pass authentication
+                    if self.config.get('username',False) or self.config.get('password',False):
+                        logger.verbose('Credential passwords s are redacted to protect the innocent...')
+                        logger.verbose('Auth credentials: username=[{}] password sha256 hash is "{}"'.format(self.config.get('username'),sha256(str(self.config.get('password','')).encode('utf-8')).hexdigest()))
+                        logger.verbose('You can validate them yourself by calculating the sha256 hex digest of your password string (google is your friend if you do not know how to do this)')
+                        logger.verbose('Note: a password that is not provided (i.e. blank) will hash to "{}"'.format(sha256(str('').encode('utf-8')).hexdigest()))
+
+                        self.username_pw_set=(self.config.get('username',None),self.config.get('password',None))
+
+                    logger.verbose("Connecting to {}:{}".format(self.config.get('broker_address'),str(self.config.get('broker_port'))))
+                    self.connect(self.config.get('broker_address'), self.config.get('broker_port'), self.config.get('broker_timeout'))
+                    logger.verbose("Connected to MQTT broker")
+
+                    logger.verbose('Publishing message [{}] to topic [{}] '.format(self.config.get('payload'),self.config.get('topic')))
+                    publish_info = self.publish(self.config.get('topic'), self.config.get('payload'), qos=self.config.get('qos'), retain=self.config.get('retain'))
+                    logger.verbose("Notification sent to broker, waiting for callback response to confirm publishing success - rc={}".format(publish_info))
+
+                    self.loop(timeout=self.config.get('broker_timeout'))
+                    self.loop_start()  #Non-blocking
+
+                    #self.loop_forever()  # blocking
+
+                except Exception as e:
```

Comment:
wrapping everything in a global try except block like this is bad practice. Try be as minimal as possible with the block

Reply:
Also fixed during code refactoring 964dba7624322c86701376d4875a23025a9b8bf8.  Only have 2 try blocks, one each around ```mqtt.connect()``` and ```mqtt.publish()```